spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesInformerDiscoveryClient.java

```diff
@@ -78,9 +78,10 @@ public KubernetesInformerDiscoveryClient(String namespace, SharedInformerFactory
 			Lister<V1Service> serviceLister, Lister<V1Endpoints> endpointsLister,
 			SharedInformer<V1Service> serviceInformer, SharedInformer<V1Endpoints> endpointsInformer,
 			KubernetesDiscoveryProperties properties) {
-		this.sharedInformerFactory = sharedInformerFactory;
-		this.serviceLister = serviceLister;
-		this.endpointsLister = endpointsLister;
+		this.sharedInformerFactories = List.of(sharedInformerFactory);
```

Comment:
a `List` with a single element here, as such, nothing changes for the clients that were already using this constructor and the functionality before "selective namespaces" were added.